db/warehouse/migrate/20241111212106_add_created_by_to_custom_assessment.rb

```diff
@@ -0,0 +1,17 @@
+#  Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+#  License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+#
+
+class AddCreatedByToCustomAssessment < ActiveRecord::Migration[7.0]
+  def change
+    safety_assured do
+      # These refer to the users table in the app db (not warehouse), so fk relationship is not made explicitly
+      add_reference :CustomAssessments, :created_by_user, null: true
+      add_reference :CustomAssessments, :updated_by_user, null: true
+      # This refers to the User table in the warehouse db
+      add_reference :CustomAssessments, :created_by_hud_user, foreign_key: { to_table: :User }, null: true
+      # no need to add updated_by_hud_user, since UserID column exists
```

Comment:
I may have missed something in the PR discussion, but why do we need `created_by_hud_user`?

TBH I would be more inclined to eventually drop the UserID column entirely, and just generate it on HMIS CSV exports (once we start exporting custom assessments...).  I don't see the benefit in keeping both references to `users` and `User` tables. (_Maybe_ there is an argument for having that reference for migrated-in data, to store a reference to some associated "user" that doesn't exist in the application).

Reply:
Ah thanks for raising this @gigxz. Theron and I discussed it on a huddle but I forgot to update the PR description to describe what we talked about.

My understanding is that the `users` table is currently difficult to import into and join within Superset, since it's in the app db, but the `User` table is already there and easily joinable, since it's in the warehouse db. The long term vision is that we don't want to keep the 2 separate dbs, so everything will be easily imported into Superset, and we would then drop the `created_by_hud_user` column in favor of `created_by_user`. 

Based on your comment, I wonder if `created_by_hud_user` is the one that should prevail, in which case `created_by_user` could be removed now?

Reply:
@ttoomey  I'd like to understand the superset limitation a little better-- is there an issue I can look at? I wonder if we can find a path forward for joining from warehouse db tables to the `users` table at least. There are other warehouse records that reference the users table, that will need to join with `users` to display information in superset. For example:
* staff assignment (reference users directly) - I can see this being asked for in Superset to track caseload over time
* files (reference users directly) - I don't see a clear use-case for reporting on files, but I'm sure there will be one eventually
* user access controls - reporting on access control changes (may or may not want to end up in superset, but tagging https://github.com/open-path/Green-River/issues/404)

Reply:
Notes from my huddle with @gigxz, to discuss at dev standup Monday:

Background info
- Right now we’re not displaying the UserID anywhere in the application.
- We used to, but now we load the user from paper trail. 
- Adding these cols was intended to just lighten the expensive load of fetching it from paper trail.
- UserID is required by HUD, but not used in reports.- HUD doesn’t even specify if it’s the user who created or last updated. Right now we update UserID anytime we update the record.

Ideal future state
- For most things that want to reference user, use application users. 
  - We own/control that table and it’s not subject to disruptions if someone’s email address changes
- What’s the future of the HUD User table? 
  - Maybe (extreme case) it even goes away, and is just generated at export when we make a CSV? (Though this is a big departure from our posture which is that the db schema matches hud exactly)

2 possible paths for this PR
- Path 1: If we can’t join with the users table. 
    - Then just point it at UserID (w/out worrying about impersonation) and don’t update when the assessment updates
    - Still add the user reference columns for future use
    - In the place where we call the HUD user the “hmis” user, rename that across the whole file
- Path 2 (preferred): How hard would it be to pursue a way to get the users table into superset, w/out merging it into the warehouse database?
    - Get rid of created_by_hud_user
    - Keep created_by_user and updated_by_user
    - Make some decision about impersonation. It probably doesn’t matter that much to customers either way

Side note about impersonation
- Use case: Someone who likes impersonating while doing data correction, so that the “real” person who created the assessment is still the user associated with it

Reply:
Decision from dev standup 11/25
- add `created_by_hud_user_id` and `updated_by_hud_user_id` both pointing at the HUD user table by PK, and not by data source relation
- don't add any columns related to application user

Reply:
thanks @martha !